### PR TITLE
Modify elasticdl job CRD sample

### PR DESCRIPTION
**Pull request by skydoorkai**
To simplify the ElasticDL job CRD.

### CODE REVIEW

File: docs/designs/elasticdl_operator.md [hunk 1]
```diff
@@ -127,73 +127,36 @@ spec:
...
-            path: /data
+    count: 10
+    priority: high=0.5
+    image: elasticdl-worker
+    resource_request:
+      cpu: 4
+      gpu: 1
+      memory: 2048Mi
+    volume:
+      host_path=/host_data,mount_path=/data
```
Comment (QiJune) [inline docs/designs/elasticdl_operator.md]: How about following the naming convention of Kubernetes pod spec.

```yaml
replicas: 10
priorityClassName: low
image: elasticdl-worker
resources:
  requests:
    cpu: 4
    gpu: 1
    memory: 2048Mi
volume: "host_path=/host_data,mount_path=/data"
```

Comment (skydoorkai) [reply docs/designs/elasticdl_operator.md]: Controller will translate these values into ElasticDL arguments, such as:
`worker_pod_priority=low`, `ps_resource_request=cpu=1,memory=4096Mi`, and the master will use Kubernetes python API.
It is the Kubernetes cluster that determine the name of `priorityClassName`. And we want to support `high=0.5` format for workers. Also, ElasticDL does not care about resource limit `ps_resource_limit` as long as the requested resources are satisfied.

Thus, changed to:

```
  Worker:
    replicas: 10
    priority: high=0.5
    image: elasticdl-worker
    resource_request:
      cpu: 4
      gpu: 1
      memory: 2048Mi
    volume: "host_path=/host_data,mount_path=/data"
```

File: docs/designs/elasticdl_operator.md [hunk 2]
```diff
@@ -119,81 +119,35 @@ metadata:
...
+    count: 10
+    priority: high=0.5
+    image: elasticdl-worker
+    resource_request: "cpu=4,gpu=1,memory=2048Mi"
+    volume: "host_path=/host_data,mount_path=/data"
 ```
 
 ## ElasticDL CRD
 
+TODO: change CRD definition to be consistent with the sample above.
```
Comment (QiJune) [inline docs/designs/elasticdl_operator.md]: Please remove the TODO comment

File: docs/designs/elasticdl_operator.md [hunk 3]
```diff
@@ -226,15 +180,28 @@ spec:
...
+                volume: string
+            Worker:
+               properties:
+                count: integer
                   minium: 1
+                image: string
+                priority: string
+                resource_request: string
+                volume: string
+              type: object
```
Comment (QiJune) [inline docs/designs/elasticdl_operator.md]: Remove this line?

**Review by QiJune — APPROVED** [review]
Excellent!

File: docs/designs/elasticdl_operator.md [hunk 4]
```diff
@@ -226,15 +181,27 @@ spec:
...
+                  minium: 0
+                image: string
+                priority: string
+                resource_request: string
+                volume: string
+            Worker:
+               properties:
+                count: integer
                   minium: 1
+                image: string
```
Comment (QiJune) [inline docs/designs/elasticdl_operator.md:201]: A quick question, both formats are legal?

```
image: string
```

```
image:
  type: string
```

**Pull request merged**